treeherder/webapp/api/intermittents_view.py

```diff
@@ -0,0 +1,124 @@
+            if len(match) == 0:
+                hash_list.append(item['job__option_collection_hash'])
+
+        hash_query = OptionCollection.objects.filter(option_collection_hash__in=hash_list).select_related(
+                                                     'option').values('option__name', 'option_collection_hash')
+
+        for item in queryset:
+            match = filter(lambda x: item['job__option_collection_hash'] == x['option_collection_hash'], hash_query)
+            if len(match) > 0:
+                item['build_type'] = match[0]['option__name']
+            else:
+                item['build_type'] = 'unknown'
+
+        return queryset
+
+
+class FailureCount(generics.ListAPIView):
+    """ List of failures (optionally by bug) and testruns by day per date range and tree"""
+
+    serializer_class = FailureCount
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
+        endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
+        tree = list(get_tree(self.request.query_params.get('tree').encode('utf-8')))
+        bug_id = self.request.query_params.get('bug')
+
+        push_query = Push.objects.filter(repository_id__in=tree,
+                                         time__range=(startday, endday)).annotate(date=TruncDate('time')).values(
+                                         'date').annotate(test_runs=Count('author')).order_by('date').values(
+                                         'date', 'test_runs')
+
+        if bug_id:
+            job_query = BugJobMap.objects.filter(job__repository_id__in=tree,
+                                                 job__push__time__range=(startday, endday),
+                                                 job__failure_classification__id=4,
+                                                 bug_id=int(bug_id)
+                                                 ).select_related('push').annotate(date=TruncDate('job__push__time'))\
+                                                 .values('date').annotate(failure_count=Count('id')).order_by(
+                                                 'date').values('date', 'failure_count')
+        else:
+            job_query = Job.objects.filter(push__time__range=(startday, endday),
+                                           repository_id__in=tree,
+                                           failure_classification_id=4).select_related('push').annotate(
+                                           date=TruncDate('push__time')).values('date').annotate(
+                                           failure_count=Count('id')).order_by('date').values('date', 'failure_count')
+
+        queryset = []
+
+        for push in push_query:
+            match = filter(lambda x: push['date'] == x['date'], job_query)
+            if len(match) > 0:
+                match[0]['test_runs'] = push['test_runs']
+                queryset.append(match[0])
```

Comment:
It's quite hard to read this to understand what this does? Could you add some comments? (They may also make it easier to suggest another way to do this)